#Publish all joint states
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 30


# Leg Springs
leg1_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg1
  pid: {p: 8.214, i: 0, d: 0.712} # 0.5476 {p: 8.214, i: 0, d: 0.712} 
leg2_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg2
  pid: {p: 8.214, i: 0, d: 0.712} 
leg3_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg3
  pid: {p: 8.214, i: 0, d: 0.712} 
leg4_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg4
  pid: {p: 8.214, i: 0, d: 0.712} 
leg5_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg5
  pid: {p: 8.214, i: 0, d: 0.712} 
leg6_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg6
  pid: {p: 8.214, i: 0, d: 0.712} 
leg7_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg7
  pid: {p: 8.214, i: 0, d: 0.712} 
leg8_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint_leg8
  pid: {p: 8.214, i: 0, d: 0.712} 

# Kinematic constraint on the wheels
wheel1_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel1
wheel2_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel2
wheel3_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel3
wheel4_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel4
wheel5_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel5
wheel6_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel6
wheel7_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel7
wheel8_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: joint_wheel8

# Manipulator controller
# Test 6 {p: 57.947, i: 20, d: 13.216} <<< for live
# Test 8 {p: 70.369, i: 25, d: 13} <<< for sim
joint1_position_controller:
  type: effort_controllers/JointPositionController
  joint: Joint_1
  pid: {p: 70.369, i: 25, d: 1}       
joint2_position_controller:
  type: effort_controllers/JointPositionController
  joint: Joint_2
  pid: {p: 70.369, i: 25, d: 1}              
joint3_position_controller:
  type: effort_controllers/JointPositionController
  joint: Joint_3
  pid: {p: 70.369, i: 25, d: 1}                 
joint4_velocity_controller:
  type: velocity_controllers/JointVelocityController
  joint: Joint_4
  pid: {p: 10, i: 0, d: 0}
